### ISSUE TITLE
Adição de mensagem de confirmação de logout

### ISSUE THREAD

**Issue by diogomene15** [diogomene15/TrabalhoFinal-GCS#11]
O projeto atual da aplicação direciona o usuário para a tela de listagem de endereços após o login bem-sucedido. No entanto, ao clicar no botão "voltar" estando nessa tela, o usuário é deslogado automaticamente, retornando à tela de login e perdendo o acesso ao sistema, o que pode ser frustrante e resultar em interrupções desnecessárias na experiência do usuário.

A falta de uma confirmação de logout pode levar a logouts acidentais, prejudicando a experiência do usuário ao interagir com a aplicação. Isso pode ocorrer especialmente em dispositivos móveis, onde o botão "voltar" é comumente utilizado para navegar entre telas, e não necessariamente para deslogar da aplicação.

Para resolver esse problema, proponho a implementação de uma mensagem de confirmação ao usuário ao clicar no botão "voltar" na tela de listagem de endereços. Essa mensagem deve solicitar que o usuário confirme sua intenção de deslogar da aplicação antes de executar essa ação. A implementação dessa confirmação não apenas previne logouts acidentais, mas também melhora a usabilidade ao garantir que ações críticas como o logout sejam deliberadas e intencionais por parte do usuário.

No arquivo `EnderecosFragment.java`, podemos adicionar um diálogo de confirmação ao sobrescrever o método `onBackPressed()`.

Recomendo o envolvimento da equipe de UX/UI para auxiliar na criação de um fluxo de uso agradável ao usuário durante a confirmação de logout. A equipe de UX/UI deve garantir que a mensagem de confirmação seja clara e compreensível, evitando qualquer ambiguidade na intenção do usuário. Além disso, é importante validar essa solução com testes de usabilidade para garantir que ela atenda às expectativas e necessidades dos usuários finais. Essa abordagem deve ser discutida e validada em conjunto com a equipe de Desenvolvimento para assegurar que todos os aspectos técnicos e de usabilidade sejam adequadamente considerados e implementados.

**Comment by diogomene15** [comment]
@JoaoPKarpinski , poderia auxiliar no atendimento dessa questão, por favor?

**Comment by JoaoPKarpinski** [comment]
Claro! Pode deixar! Acredito que a criação de um método estático em uma classe nova, que "infla" a caixa de confirmação na tela, pode ser uma boa solução com encapsulamento útil no futuro.
